game/hud/src/lib/playerStateEqual.ts

```diff
@@ -5,16 +5,26 @@
  *
  */
 
-import { PlayerState } from '@csegames/camelot-unchained/lib/core/clientInterface';
+import { SelfPlayerState, EnemyTargetState, FriendlyTargetState } from '@csegames/camelot-unchained';
 import { numEqualsCloseEnough } from '@csegames/camelot-unchained/lib/utils';
 import { BodyParts } from './PlayerStatus';
 
-export function isEqualPlayerState(a: PlayerState | GroupMemberState, b: PlayerState | GroupMemberState) {
+type PlayerState = SelfPlayerState | EnemyTargetState | FriendlyTargetState;
+
+export function isEqualPlayerState(a: PlayerState, b: PlayerState) {
```

Comment:
this function most definitely does not check if these two PlayerStates are equal.

It checks if it is a playerstate, whether one or the other is falsy and if they have matching character & entity ids.

Reply:
These are just the changes in the commit, not the whole function. The whole function checks blood, health, stamina, statuses, etc. It's very similar to what we had before, just updated with new api.